Added maven dependency for Bouncy Castle to server

lutovich opened the pull request:
It is used by Netty for self-signed certificate generation. It is a fallback path if proprietary package `sun.security.x509` provided by OpenJDK is not in the classpath. In particular, IBM JDK does not include this package.

lutovich commented:
@jakewins could you please review this PR?

jotomo commented on pom.xml:
```diff
@@ -27,6 +27,7 @@
     <lucene.version>3.6.2</lucene.version>
     <powermock.version>1.5.5</powermock.version>
     <logback-classic.version>1.1.2</logback-classic.version>
+    <bouncycastle.version>1.52</bouncycastle.version>
```
Current version is the recently released 1.53 , which has some compatibility fixes that might be of interest to the server:
`The provider has also been updated to reflect changes in JDK 1.8 which broke X509Certificate.hashCode() and X509Certificate.verify(PublicKey, Provider)` 
from: http://bouncycastle.org/latest_releases.html

lutovich replied:
Release 1.53 indeed worth upgrading to but probably in a separate PR. Here I've just extracted existing version as a parameter.

jotomo replied:
@lutovich You still have this (PR for 1.53) on your radar or a card? :-)

jakewins commented on community/bolt/transport-socket/pom.xml:
```diff
@@ -50,6 +50,15 @@
+    </dependency>
+    <dependency>
+      <groupId>org.bouncycastle</groupId>
+      <artifactId>bcpkix-jdk15on</artifactId>
```
Do we need to add it to Bolt in 2.3? Since we dislodged Bolt from the deps, we shouldn't need it right? Suggest we do a separate PR for adding it to Bolt in 3.0, since the modules have changed there. (in fact, I've got a card to remove Bolt entirely from the 2.3 branch now that 2.3.0 is out).

lutovich replied:
Bolt tests are still executed during the build. And without bouncycastle some tests fail with because neither `sun.security.x509` nor `org.bouncycastle.cert` are in the classpath.
If Bolt is going to be removed than it is better to wait with this PR and change only server pom.

jakewins replied:
Hmm. I'll write a PR right now then to remove Bolt.

On Mon, Nov 2, 2015 at 2:55 PM, Konstantin Lutovich <
notifications@github.com> wrote:

> In community/bolt/transport-socket/pom.xml
> https://github.com/neo4j/neo4j/pull/5812#discussion_r43680711:
> 
> > @@ -50,6 +50,15 @@
> >      </dependency>
> > 
> > ```
> >  <dependency>
> > ```
> > -      <groupId>org.bouncycastle</groupId>
> > -      <artifactId>bcprov-jdk15on</artifactId>
> > -    </dependency>
> > -    <dependency>
> > -      <groupId>org.bouncycastle</groupId>
> > -      <artifactId>bcpkix-jdk15on</artifactId>
> 
> Bolt tests are still executed during the build. And without bouncycastle
> some tests fail with because neither sun.security.x509 nor
> org.bouncycastle.cert are in the classpath.
> If Bolt is going to be removed than it is better to wait with this PR and
> change only server pom.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/neo4j/neo4j/pull/5812/files#r43680711.

jakewins commented:
I like @jotomo's point - if there's a Java 8 fix in the next version, seems like a good idea to bump? Since this only goes in the server, bumping the dep version shouldn't affect anyones embedded dependency trees. 

Other than that and my little question re: Bolt, LGTM, happy to merge once those two things are decided on

jakewins commented:
Bolt removal PR: https://github.com/neo4j/neo4j/pull/5822

lutovich committed:
Added maven dependency for Bouncy Castle to server

It is used by Netty for self-signed certificate generation. It is a fallback
path if proprietary package sun.security.x509 provided by OpenJDK is not in
the classpath. In particular, IBM JDK does not include this package.

lutovich commented:
@jakewins Bolt removal merged, made this PR to only add bouncycastle to the server.

tinwelint commented:
@lutovich @jakewins can we get this in now to help builds be happier?

lutovich commented:
@tinwelint @jakewins this PR is ready to be merged.
